### PR TITLE
Add scalacOptions for sourcemapping github paths

### LINKED ISSUE #705: Scala JS source maps, map to CI runner dir?
In my pet project, vite complains about source map errors. I believe one of the culprits to be smithy. 

Here's a sample .js.map file. 

```
{
"version": 3,
"file": "frontend.Clients$.js",
"mappings": "blah blah",
"sources": ["file:///C:/temp/ilsclaims/shared/src/example/Client.scala", "file:///home/runner/work/smithy4s/smithy4s/modules/http4s/src/smithy4s/http4s/SimpleProtocolBuilder.scala"],
"names": ["internal-b28b7af69320201d1cf206ebf28373980add1451", "eventClient", "internal-9fb50eb6088e86acde75de71513c77ec947f608e", "exposedEventClient", "notificationClient", "claimReportClient", "ilsopWrapperClient"],
"lineCount": 59
}

```
I think the important part, is the home/runner/work/smithy4s/etc/etc line, which obviously doesn't exist, in my local environment. Because I aggressively lurk discord, I believe the solution may look something like this; 

https://discord.com/channels/632150470000902164/635668814956068864/1060561961810677760

https://github.com/typelevel/sbt-typelevel/blob/4da726ca021ea2c6459e8286d9ec49ab7c79a883/github/src/main/scala/org/typelevel/sbt/TypelevelScalaJSGitHubPlugin.scala#L35

I believe there to be a compiler flag, which could potentially point this back to this repository, for example. I will openly admit, that my understanding of this, is somewhat superficial...

### PR DESCRIPTION
**Pull request by zetashift**
Hopefully fixes https://github.com/disneystreaming/smithy4s/issues/705

Code is an amalgamation of https://github.com/typelevel/sbt-typelevel/blob/4da726ca021ea2c6459e8286d9ec49ab7c79a883/github/src/main/scala/org/typelevel/sbt/TypelevelScalaJSGitHubPlugin.scala#L34 and https://github.com/raquo/Laminar/blob/739cc0dbab319c685f28be30a1592b7d04347842/build.sbt#L76

Feedback is very much welcome, the code could be a lot simpler like the code from Laminar.

### CODE REVIEW

**Comment by zetashift** [comment]
The CI errors out during the JS tests:

```sh
[info] [coreJS] compiling 203 Scala sources to /home/runner/work/smithy4s/smithy4s/modules/core/target/js-2.13/classes ...
[error] [coreJS] :file:/home/runner/work/smithy4s/smithy4s/.sbt/matrix/coreJS/ is not a valid URI
[error] [coreJS] one error found
[error] [coreJS] (coreJS / Compile / compileIncremental) Compilation failed
[error] Total time: 35 s, completed Jan 5, 2023, 11:58:46 PM
```

I feel like I'm missing something here with my changes, but I can' quite figure out yet why the test fail.

File: project/Smithy4sBuildPlugin.scala [hunk 1]
```diff
@@ -390,6 +391,20 @@ object Smithy4sBuildPlugin extends AutoPlugin {
...
+      val flag =
+        if (scalaVersion.value.startsWith("3")) "-scalajs-mapSourceURI:"
+        else "-P:scalajs:mapSourceURI:"
+      val localSourcesPath = baseDirectory.value.toURI
+      val headCommit = git.gitHeadCommit.value.get
+      scmInfo.value.map { info =>
+        val remoteSourcesPath =
+          s"${info.browseUrl.toString
+            .replace("github.com", "raw.githubusercontent.com")}/$headCommit"
+        s"${flag}:$localSourcesPath->$remoteSourcesPath"
```
Comment (keynmol) [inline project/Smithy4sBuildPlugin.scala:405]: you have an extra `:` here (you already have one in the flag)

Comment (keynmol) [reply project/Smithy4sBuildPlugin.scala:405]: I removed `:` from the `flag` and have tests passing locally

Comment (zetashift) [reply project/Smithy4sBuildPlugin.scala:405]: What a silly mistake :P, thank you!

**Review by Baccata — APPROVED** [review]
Thanks for this !

**Comment by Quafadas** [comment]
> Thanks for this !

@zetashift Seconded :-)!

File: project/Smithy4sBuildPlugin.scala [hunk 2]
```diff
@@ -390,6 +391,20 @@ object Smithy4sBuildPlugin extends AutoPlugin {
   }
 
   lazy val jsDimSettings = simpleJSLayout ++ Seq(
+    scalacOptions ++= {
+      // Map the sourcemaps to github paths instead of local directories
+      val flag =
+        if (scalaVersion.value.startsWith("3")) "-scalajs-mapSourceURI"
+        else "-P:scalajs:mapSourceURI"
+      val localSourcesPath = baseDirectory.value.toURI
```
Comment (armanbilge) [inline project/Smithy4sBuildPlugin.scala:399]: This should be `(LocalRootProject / baseDirectory).value.toURI.toString`, exactly like it is in sbt-typelevel.

https://github.com/typelevel/sbt-typelevel/blob/4da726ca021ea2c6459e8286d9ec49ab7c79a883/github/src/main/scala/org/typelevel/sbt/TypelevelScalaJSGitHubPlugin.scala#L40

Comment (zetashift) [reply project/Smithy4sBuildPlugin.scala:399]: You're my hero.
I'll try to get a PR that fixes it today!

Comment (kubukoz) [reply project/Smithy4sBuildPlugin.scala:399]: thanks @armanbilge <3

**Pull request merged**